bot/__main__.py

```diff
@@ -21,25 +44,23 @@ async def main() -> None:
         _intents.message_content = True
         _intents.members = True
 
-        redis_session = RedisSession(
-            address=(constants.RedisConfig.host, constants.RedisConfig.port),
-            password=constants.RedisConfig.password,
-            minsize=1,
-            maxsize=20,
-            use_fakeredis=constants.RedisConfig.use_fakeredis,
-            global_namespace="sir-robin"
-        )
-
-        await redis_session.connect()
-
-        bot.instance = SirRobin(
-            redis_session=redis_session,
-            command_prefix=constants.Client.prefix,
-            activity=discord.Game("The Not-Quite-So-Bot-as-Sir-Lancebot"),
-            intents=_intents,
+        allowed_roles = (
+            constants.Roles.events_lead,
+            constants.Roles.code_jam_event_team,
+            constants.Roles.code_jam_participants
         )
-        async with bot.instance:
-            bot.instance._guild_available = asyncio.Event()
-            await bot.instance.start(Client.token)
+        async with aiohttp.ClientSession() as session:
+            bot.instance = SirRobin(
+                redis_session=await _create_redis_session(),
+                http_session=session,
+                guild_id=constants.Client.guild,
+                allowed_roles=allowed_roles,
+                command_prefix=constants.Client.prefix,
+                activity=discord.Game("The Not-Quite-So-Bot-as-Sir-Lancebot"),
+                intents=_intents,
+            )
+            async with bot.instance:
+                bot.instance._guild_available = asyncio.Event()
```

Comment:
Why is this needed here? It gets set to this in BotBase.setup_hook

Reply:
Right, sorry. Patched in 443c2c9